### PR TITLE
fix: nas dtrain bug

### PR DESCRIPTION
**Pull request by naren-determined**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/exec/worker_process.py [hunk 1]
```diff
@@ -58,4 +58,8 @@ def main() -> None:
 
 
 if __name__ == "__main__":
-    main()
+    try:
+        main()
+    except det.InvalidHP:
+        logging.info("InvalidHP detected, gracefully exiting trial")
```
Comment (rb-determined-ai) [inline harness/determined/exec/worker_process.py:64]: maybe reword this to not say the whole trial is exiting?

Comment (naren-determined) [reply harness/determined/exec/worker_process.py:64]: Well the whole trial is exiting, but the experiment is continuing?

Comment (rb-determined-ai) [reply harness/determined/exec/worker_process.py:64]: Only the worker is exiting when this gets printed.  If you have 64 workers it makes sense to see 64 'worker exiting' messages, but not to see 64 "gracefully exiting trial messages", only to then see a decidedly non-graceful exit occur.

**Review by rb-determined-ai — APPROVED** [review]
